lib/blog.js

```diff
@@ -16,115 +22,72 @@ function parseMarkdown(markdown) {
   })
 }
 
-const POSTS_PATH = path.join(__dirname, '..', 'data', 'blog')
-class BlogPost {
-  static async getAll(language) {
-    const files = await fs.readdir(POSTS_PATH)
-    const slugs = files.map((file) => path.basename(file, '.md'))
-    return slugs.map((slug) => BlogPost.get(slug, language))
-  }
-
-  static get(slug, language) {
-    return new BlogPost(slug, language)
-  }
-
-  constructor(slug, language) {
-    this._cache = new Map()
-    this._slug = slug
-    this._language = language
-  }
-
-  exists() {
-    return fs.stat(path.join(POSTS_PATH, this.filename()))
-  }
-
-  async cache(key, computation) {
-    if (this._cache.has(key)) {
-      return this._cache.get(key)
-    } else {
-      const value = await computation()
-      this._cache.set(key, value)
-      return value
-    }
-  }
-
-  _getRawMarkdown() {
-    // This block looks not very good, but they give confidence,
-    // and security that we do not get the crash when the content
-    // not synchronized in the i18n, e.g. when we publish a new post,
-    // languages that don't have the copy on the own language get the
-    // version from the disk.
-    //
-    // Why not use the English version from the i18n module?
-    // We get the desync between this repo and i18n, the save
-    // and simple way it's using the good saved copy from the disk.
-    //
-    // drink some water, be kind to yourself and others.
-
-    // If you ask why this code block is disabled?
-    // So, the Electron translations for blogs is so bad,
-    // that it's better to currently disable this and not search
-    // for fixes. The comment above this still actual for this code block.
-    // if (this._language) {
-    //   const blog = i18n.blogs[this._language][this.href()]
-    //   if (blog) {
-    //     return this.cache('_getRawMarkdown', () => blog.content)
-    //   }
-    // }
+/**
+ * @param {string} markdown
+ * @returns {graymatter.GrayMatterFile<string>}
+ */
+function parseFrontMatter(markdown) {
+  return graymatter(markdown, {
+    excerpt: true,
+    excerpt_separator: '---',
+  })
+}
 
-    const fullPath = path.join(POSTS_PATH, this.filename())
-    return this.cache('_getRawMarkdown', () => fs.readFile(fullPath))
+/**
+ * @returns {Map}
+ */
+async function getAllPosts() {
+  if (blogPosts.size !== 0) {
+    return blogPosts
   }
 
-  async _frontmatter() {
-    const markdown = await this._getRawMarkdown()
-    return this.cache('frontmatter', () => {
-      return graymatter(markdown, {
-        excerpt: true,
-        excerpt_separator: '---',
-      })
-    })
-  }
+  const files = await fs.readdir(POSTS_PATH)
+  const slugs = files.map((file) => path.basename(file, '.md'))
 
-  async frontmatter(key) {
-    const fm = await this._frontmatter()
-    return fm.data[key]
+  for (const slug of slugs) {
+    blogPosts.set(slug, await getPost(slug))
```

Comment:
I *think* we should be checking the return value for truthiness before adding it here. If the file read fails, this can return null and then we have a map with a valid slug key and and a null value. That will cause a crash later in

```js
    const posts = Array.from(blogPosts.values())
    postsInOrder = posts.sort((a, b) => b.date.localeCompare(a.date))
```

We could filter out null values there, but it's cleaner IMO to not return empty values in the first place.

NB I haven't read the code all the way through. If there's some need to keep the invalid slugs in the map elsewhere in the code, then maybe the `posts` array should just be filtered in that snippet above.